acceptance: disarm a for loop

tbg opened the pull request:
In a [recent PR], an unrelated programming error caused one node to crash
while the test was trying to query it. This lead to a runaway loop instead
of the test failing.

Confine the offending loop to a maximum duration of one minute plus make
it use backoffs.

[recent PR]: #19802

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/19842)
<!-- Reviewable:end -->

benesch reviewed (approved):
Heh, nice. LGTM.

jordanlewis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19842#-:-KyHxIn_XwX-W6rSmWj8:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tbg committed:
acceptance: disarm a for loop

In a [recent PR], an unrelated programming error caused one node to crash
while the test was trying to query it. This lead to a runaway loop instead
of the test failing.

Confine the offending loop to a maximum duration of one minute plus make
it use backoffs.

[recent PR]: #19802

benesch commented:
omg this flaked on TestDockerCSharp. I'd just god merge.

tschottdorf commented:
TestDockerCSharp is now skipped and I just rebased this, so it should pass now.
